src/main/java/core/basesyntax/ShopServiceImpl.java

```diff
@@ -0,0 +1,17 @@
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler handler = operationStrategy.getHandler(transaction.getOperation());
+            handler.apply(transaction);
```

Comment:
When calling the `apply` method on an `OperationHandler`, it's important to handle the possibility of an exception being thrown, especially when dealing with a `Purchase` operation. Make sure to check the result balance before saving it in the Storage and throw a RuntimeException if the balance is negative.